### ISSUE TITLE
jkube.build.strategy=jib does not create a layered image like jib-maven-plugin does

### ISSUE THREAD

**Issue by philipreimer** [eclipse/jkube#589]
### Info

* Eclipse JKube version : kubernetes-maven-plugin 1.1.1
* Maven version (`mvn -v`) : 
```
Apache Maven 3.6.3 (NON-CANONICAL_2019-11-27T20:26:29Z_root)
Maven home: /opt/maven
Java version: 11.0.10, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-11-openjdk
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux", version: "5.10.15-1-manjaro", arch: "amd64", family: "unix"
```
* Kubernetes / Red Hat OpenShift setup and version : not relevant, concerns only k8s:build

* If it's a bug, how to reproduce :
1. clone https://github.com/eclipse/jkube/tree/v1.1.1/quickstarts/maven/spring-boot-with-jib
2. run `mvn k8s:build -PJib-Zero-Config` as stated in the `README` 

This will produce a `target/docker/maven/eclipse-jkube-sample-spring-boot-jib/1.1.1/build/Dockerfile` like this:

```
FROM quay.io/jkube/jkube-java-binary-s2i:0.0.9
ENV JAVA_APP_DIR=/deployments
LABEL ... (left out for brevity) ...
EXPOSE 8080 8778 9779
COPY /deployments /deployments/
``` 

So it expects a complete spring-boot far.jar in `/deployments`.

But one of the main advantages of jib is that it can produce layered images of your spring boot application. See [the official jib maven spring boot example](https://github.com/GoogleContainerTools/jib/tree/master/examples/spring-boot), which creates three layers (libs, resources, classes) in the resulting docker image. Thus, if you only change your Java code and build a new image, only the last layer needs to be changed - all the others can be re-used, which will speed up builds.
Also you don't need to run `mvn package` to create a jar before building the image.

I would expect the kubernetes-maven-plugin to behave the same way when using `jkube.build.strategy=jib`. Otherwise, what's the purpose of having the jib build strategy at all?

**Sample Reproducer Project:**
* https://github.com/eclipse/jkube/tree/v1.1.1/quickstarts/maven/spring-boot-with-jib
* https://github.com/GoogleContainerTools/jib/tree/master/examples/spring-boot

**Comment by manusa** [comment]
Similar to #425

**Comment by manusa** [comment]
Should be fixed by #724
Will be available in next 1.4.0 release

**Comment by manusa** [comment]
We just released [v1.4.0](https://github.com/eclipse/jkube/releases/tag/v1.4.0), hopefully this issue should be covered. I'm closing it, in case there's something missing, don't hesitate to open a new issue so we can address it.
